cli/lsp/npm.rs

```diff
@@ -4,29 +4,32 @@ use dashmap::DashMap;
 use deno_core::anyhow::anyhow;
 use deno_core::error::AnyError;
 use deno_core::serde_json;
-use deno_npm::registry::NpmPackageInfo;
 use deno_runtime::permissions::PermissionsContainer;
 use deno_semver::package::PackageNv;
 use deno_semver::Version;
 use serde::Deserialize;
 use std::sync::Arc;
 
-use crate::args::npm_registry_default_url;
+use crate::args::npm_registry_url;
 use crate::file_fetcher::FileFetcher;
+use crate::npm::NpmFetchResolver;
 
 use super::search::PackageSearchApi;
 
 #[derive(Debug, Clone)]
 pub struct CliNpmSearchApi {
   file_fetcher: FileFetcher,
+  resolver: Arc<NpmFetchResolver>,
   search_cache: Arc<DashMap<String, Arc<Vec<String>>>>,
   versions_cache: Arc<DashMap<String, Arc<Vec<Version>>>>,
 }
```

Comment:
Just noticed this, but I think it would be more appropriate to remove all the `Arc`s and remove `Clone` on this:

```rs
#[derive(Debug)]
pub struct CliNpmSearchApi {
  file_fetcher: FileFetcher,
  resolver: NpmFetchResolver,
  search_cache: DashMap<String, Arc<Vec<String>>>,
  versions_cache: DashMap<String, Arc<Vec<Version>>>,
}
```

Then put `CliNpmSearchApi` into an `Arc` if necessary at a higher level. Internally I don't believe there's any reason to have all these properties as `Arc`s so that should probably be left to a higher level.